lib/shared-functions.sh

```diff
@@ -19,6 +19,54 @@ function read_image_version() {
   IMAGE_VERSION_PATCH=${BASH_REMATCH[3]}
 }
 
+function read_mongo_version() {
+  local mongo_image=$(read_configuration "MONGO_IMAGE")
+  local mongo_version=$(read_configuration "MONGO_VERSION")
+  if [ -z "${mongo_version}" ]; then
+    if [[ "$mongo_image" =~ ^mongo:([0-9]+)\.(.*)$ ]]; then
+      # when running a chain of commands (example: bin/up -> bin/docker-compose) we're passing
+      # SKIP_WARNINGS=true to prevent the warning message to be printed several times
+      if [[ -z ${SKIP_WARNINGS:-} ]]; then
+        echo "-------------------  WARNING  ----------------------"
+        echo "  Deprecation warning: the mongo image is now split between MONGO_IMAGE"
+        echo "  and MONGO_VERSION configurations. Please update your config/overleaf.rc as"
+        echo "  your current configuration may stop working in future versions of the toolkit."
+        echo "  Example: MONGO_IMAGE=mongo"
+        echo "           MONGO_VERSION=6.0"
+      fi
+      MONGO_VERSION_MAJOR=${BASH_REMATCH[1]}
+      MONGO_DOCKER_IMAGE="$mongo_image"
+    else
+      echo "---------------------  ERROR  -----------------------"
+      echo "  The mongo image is now split between MONGO_IMAGE and MONGO_VERSION configurations."
+      echo "  Please update your config/overleaf.rc."
+      echo ""
+      echo "  MONGO_VERSION must start with the actual major version of mongo, followed by a dot."
+      echo "  Example: MONGO_IMAGE=my.dockerhub.com/mongo"
```

Comment:
```suggestion
      echo "  Example: MONGO_IMAGE=my.dockerhub.com/custom-mongo"
```
Bonus: WDYT about aligning the two examples?

Reply:
Sure, done. My idea was to align with the situation causing the error, rather than with other messages, but there's no much difference and may be less confusing for devs.